opencensus/metrics_listener.go

```diff
@@ -0,0 +1,48 @@
+package opencensus
+
+import (
+	"time"
+
+	"go.opencensus.io/stats"
+
+	gosundheit "github.com/AppsFlyer/go-sundheit"
+)
+
+// MetricsListener reports metrics on each check registration, start and completion event (as gosundheit.CheckListener)
+// This listener all reports metrics for the entire service health (as gosundheit.HealthListener)
+type MetricsListener struct {
+	classification string
+}
+
+func NewMetricsListener(opts ...Option) *MetricsListener {
```

Comment:
No need for a pointer here IMHO, the struct is not mutated.

Reply:
The pointer here is so it will actually return the interface

Reply:
I'm not sure I understand. If you remove the pointer receiver from everywhere it'll implement the interface.

Reply:
If you only use the struct, you will need to call the `WithCheckListener` in less fluent way (you're still required to pass the pointer to method receiving the interface) - so easing the usage and returning a pointer.

Reply:
> you're still required to pass the pointer to method receiving the interface

My suggestion is to remove the pointer from the method receivers as well. There is no state mutation in this struct, using a pointer is unnecessary and has nothing to do with interface implementation.